go.mod

```diff
@@ -20,14 +22,14 @@ require (
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.8.4
 	gomodules.xyz/jsonpatch/v2 v2.4.0
-	helm.sh/helm/v3 v3.13.3
-	k8s.io/api v0.28.5
-	k8s.io/apiextensions-apiserver v0.28.5
-	k8s.io/apimachinery v0.28.5
-	k8s.io/cli-runtime v0.28.5
-	k8s.io/client-go v0.28.5
-	k8s.io/utils v0.0.0-20230505201702-9f6742963106
-	sigs.k8s.io/controller-runtime v0.16.3
+	helm.sh/helm/v3 v3.14.1
+	k8s.io/api v0.29.0
+	k8s.io/apiextensions-apiserver v0.29.0
+	k8s.io/apimachinery v0.29.0
+	k8s.io/cli-runtime v0.29.0
+	k8s.io/client-go v0.29.0
+	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
+	sigs.k8s.io/controller-runtime v0.17.2
 	sigs.k8s.io/kubebuilder/v3 v3.13.0
```

Comment:
Can we bump kubebuilder to 3.14.0 as well? This is the version that Kubebuilder bumped to k8s 1.29 support and will make it easier when trying to pull a new version of the helm-operator-plugins into the operator-sdk for the SDK k8s bump